Sources/TuistKit/Cache/CacheController.swift

```diff
@@ -180,6 +185,24 @@ final class CacheController: CacheControlling {
         }
     }
 
+    private func getMacCatalystTargets(graph: Graph, scheme: Scheme) -> [Target]? {
```

Comment:
Do we need to precompute the array? Or can't we instead have an extension on `Target`? 
```swift
extension Target {
  var supportsCatalyst: Bool { switch ... }
}
```

I feel that would make the implementation more straightforward. The extension can be file-scoped to the cache controller in case we don't want to pollute `Target`

Reply:
In my opinion, the reason we need to create the `Target` array in `CacheController` is that if a project contains frameworks that don't support MacCatalyst, and if `CacheController` tries to send a boolean flag instead of creating an array, the cache won't be generated if all targets don't support Catalyst. For projects that support Catalyst, a considerable portion of them will likely include frameworks that don't support Catalyst.

I've simulated this situation using SPM and created a test project accordingly. Below is the sample project.

[MacCatalystExampleProject with SPM](xcode://clone?repo=https%3A%2F%2Fgithub.com%2Fdevyhan%2FMacCatalystExample)

However, using `Target` array approach with last commit, when generating a cached project and building it with the `tuist generate --xcframeworks` command, problems arise if the build cache includes frameworks that don't support Catalyst but the project has frameworks that do support Catalyst. This seems like a separate issue from the one being discussed, and it appears to be related to something similar to #4610. Below is the error that occurs when building with a cache that includes Catalyst unsupported frameworks.

<img width="423" alt="Pasted Graphic 21" src="https://user-images.githubusercontent.com/45344633/227095767-fac4aff2-f3d9-4b29-8bfd-ceed0a3ae75e.png">

Is there a better approach?

Reply:
I don't think I follow – how is the result different if we basically compute the array inside `bundleArtifactBuilder`'s `build` function instead of `CacheController`? Those two things should be equivalent 🤔

Reply:
I was mistaken in my thinking and made a simple change to the implementation without precomputing the array. Additionally, I have improved the readability of the `switch` statement that checks for Catalyst support by using `contains`.